v2/middlewares/measurement.go

```diff
@@ -0,0 +1,69 @@
+package middlewares
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+
+	"github.com/honestbank/kp/v2/internal/middleware"
+)
+
+var operationDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Name:    "kp_operation_time_milliseconds",
+	Buckets: []float64{1, 5, 50, 200, 500, 1_000, 2_000, 5_000, 15_000, 45_000},
+}, []string{"result", "error"})
+
+var totalProcessed = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "message_processed_total",
+		Help: "Number of messages processed.",
+	},
+	[]string{"result", "error"},
+)
+
+type measurementMiddleware struct {
+	pushClient *push.Pusher
+}
+
+func (m measurementMiddleware) SetupBackgroundJob() {
+	go func() {
+		err := m.pushClient.Push()
```

Comment:
Isn't this supposed to be looping? as to push continiously? Or I misunderstood how the API of the middleware should work, but just in case, confirm this is correct, to just push one. i.e. is this going to be invoked in every kafka message processed, or only once?

Reply:
you're right

Reply:
pushed :)